DatWeatherDoe/UI/Decorator/Text/RiseSet/RiseSetTextBuilder.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  RiseSetTextBuilder.swift
+//  DatWeatherDoe
+//
+//  Created by Markus Markus on 7/26/22.
+//  Copyright © 2022 Markus Mayer.
+//
+
+import Foundation
+
+final class RiseSetTextBuilder {
+
+    private let initial: String
+    private let sunset: TimeInterval
+    private let sunrise: TimeInterval
+
+    private let upArrowStr = "⬆"
+    private let downArrowStr = "⬇"
+
+    init(
+        initial: String = "",
+        sunset: TimeInterval = 0,
+        sunrise: TimeInterval = 0
+    ) {
+        self.initial = initial
+        self.sunset = sunset
+        self.sunrise = sunrise
+    }
+
+    func build() -> String {
+        let riseSetString = buildRiseSet()
+        if riseSetString == "" {
+            return initial
+        }
+        return (initial != "") ? "\(initial) \(riseSetString)" : riseSetString
```

Comment:
Could refactor this as follows for readability:

```
if let initial = initial {
    return "\(initial) \(riseSetString)"
}
return riseString
```

Reply:
I found that this proposed change doesn't work ("Initializer for conditional binding must have Optional type, not 'String'"). Also, I don't think it would do the right thing.

It's all about ensuring there isn't a surplus space. If riseSetString is empty, return initial. If initial is empty return riseSetString. Otherwise concatenate them with a space in between. So, maybe I should do that instead?

if riseSetString.isEmpty {
    return initial
}
if initial.isEmpty {
    return riseSetString}
}
return [initial, riseSetString].joined(separator: " ")